java/clients/src/test/java/sleeper/clients/QueryWebSocketClientTest.java

```diff
@@ -280,7 +280,8 @@ void shouldHandleMessageWithUnrecognisedType() throws Exception {
             // When / Then
             assertThat(runQueryFuture(query,
                     withResponses(
-                            message(unknownMessage("test-query-id")))))
+                            message(unknownMessage("test-query-id")),
+                            close("Connection closed normally"))))
```

Comment:
Thanks, this does look like a bug since the connection closed last time when message validation failed. This was causing the base PR to hang when running queries so I've fixed this in the base PR